### ISSUE TITLE
[RFC]: Breaking changes for country codes related to spatial aggregation.

### ISSUE THREAD

**Issue by VIKTORVAV99** [electricitymaps/electricitymaps-contrib#4733]
In order to aggregate some countries under the country level country code we need to free up the current country level ISO code for countries that have subzones yet still use the country level identifier for the "main" subzone.

So I propose we switch Spain for example from using the ES identifier to using the ES-ES and France from FR to FR-FR and so on.

While doing this we should also rename the Australian zones to match the 2 letter ISO code we use for the other countries, so from AUS to AU.

Doing these changes will allow us to put the aggregated countries under the "main" ISO country code which currently is a blocker for aggregating Spain and France and will be a blocker for GB once we have enough data to aggregate it.

**Comment by madsnedergaard** [comment]
cc @tonypls, I think this makes good sense - any thoughts on this? 
The only issue I see is people having bookmarked the app on the zone (/FR), but I guess they'll just see the aggregated county instead of the specific zone (which was the whole country when they bookmarked it anyway).

**Comment by VIKTORVAV99** [comment]
Yeah for France as an example it would just include Corsica as well if we follow the current "rules" (don't aggregate autonomous regions with their own ISO country codes, Åland, Greenland, etc) and wouldn't create a massive impact on the overall carbon intensity.

I'm open to alternatives but this seems like the easiest option that's provided users with the experience they would expect. (going to the country key shows the whole country).

**Comment by VIKTORVAV99** [comment]
> I think it's a pretty good approach and definitely think we should trim AUS down to AU. 
> 
> I do find ES-ES and FR-FR a little strange though. Maybe we should follow our existing pattern and make mainland Spain ES-ES1 and France FR-FR1.

My thinking was that we use zone codes that normally wouldn't be used so we lower the risk of zone conflicts in the future. For example there are talks about splitting up France and if we used FR-FR as our zone key it would be no risk of "zone key collision" if France is split up savings us a lot of work in the future.

**Comment by VIKTORVAV99** [comment]
@tonypls @madsnedergaard anything else to discuss here or can I create a PR with all breaking changes?
Only thing I can think of would be the API endpoints would change as well and that would need to be comunicated if I'm not wrong?

**Comment by madsnedergaard** [comment]
Hi again, we've been discussing this and it's definitely something we need to address when we aggregate these edge-cases! 

However, this will mean a breaking change for our commercial API and due to #4737 and #4731  we are very limited on capacity right now.

There are also other options to explore, so we'd like to park this issue for now and return to it later on when we're ready to aggregate one of these zones :)

**Comment by VIKTORVAV99** [comment]
I was afraid it would break the API.

We do have some zones ready to be aggregated right away that would need these changes like Spain and France where we have all the data for all zones that use the ES-* and FR-* zone keys in line with the current aggregations, but let's park it for now and focus on the US, Brazil and the Faroe Islands.

There is also #4677 that is blocking all these aggregations (except maybe Faroe Islands as it has no exchanges).

We have plenty of stuff to do in the meantime.

**Comment by VIKTORVAV99** [comment]
We are closing down some longstanding issues that have't had any activity in a while. If you believe this is still an issue let us know and we can reopen it.
